src/wasm/wasm-binary.cpp

```diff
@@ -6043,14 +6053,11 @@ void WasmBinaryBuilder::visitRefIs(RefIs* curr, uint8_t code) {
 void WasmBinaryBuilder::visitRefFunc(RefFunc* curr) {
   BYN_TRACE("zz node: RefFunc\n");
   Index index = getU32LEB();
-  if (index >= functionImports.size() + functionSignatures.size()) {
-    throwError("ref.func: invalid call index");
-  }
```

Comment:
We still need this check, I think. The binary can contain an invalid index.

Reply:
The call to `getTypeByFunctionIndex` below will check that the index is valid.

Reply:
Ah, I see. That is slightly surprising, in particular, since `functionRefs[index]` seems like it could access memory out of bounds - if it was an array initialized to the number of functions. How about adding a comment like
```
// Note that we do not need to check that 'index' is in bounds, as that will be
// verified lower down. (Also, note that functionRefs[index] may write to an odd
// place in the functionRefs map if index is invalid, but that can cause no
// harm.)
```
?